pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -299,10 +299,14 @@ func (r *ReconcileNode) reconcileCrashRetention(namespace string, cephCluster ce
 		var cronJob client.Object
 		// minimum k8s version required for v1 cronJob is 'v1.21.0'. Apply v1 if k8s version is at least 'v1.21.0', else apply v1beta1 cronJob.
 		if useCronJobV1 {
-			// delete v1beta1 cronJob if it already exists
-			err = r.client.Delete(r.opManagerContext, &v1beta1.CronJob{ObjectMeta: objectMeta})
-			if err != nil && !kerrors.IsNotFound(err) {
-				return errors.Wrapf(err, "failed to delete CronJob v1beta1 %q", prunerName)
+			// delete v1beta1 cronJob if v1Job is not found
+			if _, err := r.context.Clientset.BatchV1().CronJobs(objectMeta.Namespace).Get(r.opManagerContext, prunerName, metav1.GetOptions{}); err != nil {
```

Comment:
How about factoring out to a helper method to delete the cronjob? This looks duplicated.